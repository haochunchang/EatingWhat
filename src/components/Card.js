import React from "react";

const Card = (props) => {
  const { url, name, category } = props;
  return (
    <div className="tc bg-light-green dib br5 pa3 ma3 grow bw2 shadow">
      <img alt="food" src={url} />
      <div>
        <p>{category}</p>
        <p>{name}</p>
      </div>
    </div>
  );
};

export default Card;
