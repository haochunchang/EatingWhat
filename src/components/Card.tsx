import React from "react";

interface CardStatelessProps {
  url: string;
  name: string;
}

const Card: React.FunctionComponent<CardStatelessProps> = ({ url, name }) => {
  return (
    <div className="tc dib br5 pa3 ma3 grow bw2 shadow">
      <a
        href={`https://www.google.com.tw/maps/search/${name}/`}
        target="_blank"
        rel="noopener noreferrer"
      >
        <img alt={name} src={url} />
      </a>
      <div>
        <h3>{name}</h3>
      </div>
    </div>
  );
};

export default Card;
