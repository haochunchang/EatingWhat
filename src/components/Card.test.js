import React from "react";
import { shallow } from "enzyme";
import Card from "./Card";

describe("Card", () => {
  const mockProps = {
    name: "sushi",
    url: "www.sushi.com",
  };
  it("matches Card component snapshot", () => {
    expect(shallow(<Card props={mockProps} />)).toMatchSnapshot();
  });
});
