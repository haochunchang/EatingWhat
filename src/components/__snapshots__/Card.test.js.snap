// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches Card component snapshot 1`] = `ShallowWrapper {}`;
